### PR TITLE
Marine DA updates

### PR DESCRIPTION
**Pull request by guillaumevernieres**
# Description
Companion PR to [GDASApp/pull/1226](https://github.com/NOAA-EMC/GDASApp/pull/1226).

waiting for [global-workflow/issues/2797](https://github.com/NOAA-EMC/global-workflow/issues/2797)

Addresses a few issues described in:

- [GDASApp/issues/1219](https://github.com/NOAA-EMC/GDASApp/issues/1219)
- [GDASApp/issues/1217](https://github.com/NOAA-EMC/GDASApp/issues/1217)
- [GDASApp/issues/1232](https://github.com/NOAA-EMC/GDASApp/issues/1232)

# Type of change
- Bug fixes (fixes something broken)
- New features (adds functionality)
- Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO

# How has this been tested?
Tested on hera at c48 and c384 within the `gdas.cd` test and a subset of the `global-workflow` ci tests

# Checklist
- [ ] Any dependent changes have been merged and published
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [ ] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

File: jobs/rocoto/prepoceanobs.sh [hunk 1]
```diff
@@ -14,8 +14,10 @@ export jobid="${job}.$$"
 
 ###############################################################
 # setup python path for class defs and utils
-
-PYTHONPATH="${PYTHONPATH:+${PYTHONPATH}:}${HOMEgfs}/ush"
+# shellcheck disable=SC2311
+pyiodaPATH="${HOMEgfs}/sorc/gdas.cd/build/lib/python$(detect_py_ver)/"
```
Comment (DavidHuber-NOAA) [inline jobs/rocoto/prepoceanobs.sh:18]: Would it make more sense to link this path into `${HOMEgfs}/ush/pygfs` as part of `link_workflow.sh`?

**Review by aerorahul — DISMISSED** [review]

File: jobs/rocoto/prepoceanobs.sh (hunk 1 above)
Comment (guillaumevernieres) [reply jobs/rocoto/prepoceanobs.sh:18]: It seems fine to me as is @DavidHuber-NOAA . This is just copy/pasted from the other "prep" obs scripts, see ``` prepatmiodaobs.sh``` and ```prepsnowobs.sh```

Comment (DavidHuber-NOAA) [reply jobs/rocoto/prepoceanobs.sh:18]: Alright, I'm good with this solution.

**Review by DavidHuber-NOAA — DISMISSED** [review]

**Comment by guillaumevernieres** [comment]
I just updated the `gdas.cd` commit @DavidHuber-NOAA , this PR is ready to be tested.

**Comment by guillaumevernieres** [comment]
... Ho wait ... [global-workflow/issues/2797](https://github.com/NOAA-EMC/global-workflow/issues/2797) needs to be sorted out before we test.

File: jobs/rocoto/prepoceanobs.sh (hunk 1 above)
Comment (aerorahul) [reply jobs/rocoto/prepoceanobs.sh:18]: @guillaumevernieres This is fine for now, but will not be acceptable for ops. We will need to find alternative solutions that may need either converting this into a python package installed in a virtual environment or something else.

**Comment by guillaumevernieres** [comment]
@DavidHuber-NOAA , the fix files have been updated on hera (updates on other hpc is in progress).

**Comment by DavidHuber-NOAA** [comment]
Sounds good.  Let's wait for #2804 to help with CI testing.

**Comment by WalterKolczynski-NOAA** [comment]
@guillaumevernieres If you update your branch, I think we can launch CI once Hera comes back.

**Comment by guillaumevernieres** [comment]
I'm going to be out of town for a few weeks @WalterKolczynski-NOAA . Any chance you could just merge develop next time it's a blocker for running the ci?

**Comment by WalterKolczynski-NOAA** [comment]
> I'm going to be out of town for a few weeks @WalterKolczynski-NOAA . Any chance you could just merge develop next time it's a blocker for running the ci?

Sure. We've had people complain in the past about modifying their branches, so we try not to do it unprompted.

**Review by WalterKolczynski-NOAA — DISMISSED** [review]

**Pull request merged**